express-post-json/server.ts

```diff
@@ -0,0 +1,66 @@
+import express from 'express';
+
+const app = express();
+
+app.use(express.json());
+
+type Grade = {
+  id: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+let nextId = 1;
+
+const grades: Record<number, Grade> = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95,
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100,
+  },
+  273: {
+    id: 273,
+    name: 'Håkon Wium Lie',
+    course: 'CSS',
+    score: 92,
+  },
+};
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray: Grade[] = [];
+  for (const grade in grades) {
+    gradesArray.push(grades[grade]);
+  }
+  res.json(gradesArray);
+});
+
+app.post('/api/grades', (req, res) => {
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    return res
+      .status(404)
```

Comment:
If the client sends bad information in the body, you would usually respond with 400 instead of 404. 404 means "I can't find the endpoint you requested." 400 means "I found the endpoint but your request wasn't formulated correctly".